sys/net/application_layer/sock_dns/dns.c

```diff
@@ -67,6 +71,25 @@ void auto_init_sock_dns(void)
 
 int sock_dns_query(const char *domain_name, void *addr_out, int family)
 {
+#ifdef MODULE_SOCK_DNS_MOCK
+    if (strcmp(domain_name, SOCK_DNS_MOCK_EXAMPLE_COM_HOSTNAME)) {
+        return -ENOTSUP;
+    }
+
+    switch (family) {
+    case AF_INET:
+        memcpy(addr_out, &sock_dns_mock_example_com_addr_ipv4, sizeof(ipv4_addr_t));
+        return sizeof(ipv4_addr_t);
+
+    case AF_UNSPEC:
+        /* fall-through */
+    case AF_INET6:
+        memcpy(addr_out, &sock_dns_mock_example_com_addr_ipv6, sizeof(ipv6_addr_t));
+        return sizeof(ipv6_addr_t);
+    default:
+        return -EAFNOSUPPORT;
+    }
+#else
```

Comment:
Please don't - that just clutters the code and will make working on it harder.
You can have `sys/net/application_layer/sock_dns_mock` for an alternative implementation of the API.

Reply:
It's a real module now